tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -111,11 +111,13 @@ private protected Tracer(TracerManager tracerManager)
         }
 
         /// <summary>
-        /// Gets or sets the global <see cref="Tracer"/> instance.
+        /// Gets the global <see cref="Tracer"/> instance.
         /// Used by all automatic instrumentation and recommended
         /// as the entry point for manual instrumentation.
         /// </summary>
-        public static Tracer Instance
+        public static ITracer Instance => InternalInstance;
+
+        internal static Tracer InternalInstance
```

Comment:
I'm not sure this change is necessary or adds anything 🤔  I don't think having a `public static Tracer Instance` is fine? I can't see the benefit of the change, and IMO it just adds confusion for us?

Reply:
For the record, my reason for introducing the `ITracer` interface is so that people can do this setup in their DI config:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddSingleton<ITracer>(() => Tracer.Instance);
}
```

And use it as 

```csharp
public class MyController
{
    private readonly ITracer tracer;
    public MyController(ITracer tracer)

    public string HelloWorld()
    {
        using var scope = tracer.StartActive("Something);
        return "HelloWorld!";
    }
}
```

That way they can mock `ITracer` in their tests